package/src/components/Channel/Channel.tsx

```diff
@@ -817,14 +877,18 @@ const ChannelWithContext = <
   /**
    * Subscription to the Notification mark_read event.
    */
-  useEffect(() => {
-    const handleEvent: EventHandler<StreamChatGenerics> = (event) => {
-      if (channel.cid === event.cid) copyChannelState();
-    };
+  useEffect(
+    () => {
+      const handleEvent: EventHandler<StreamChatGenerics> = (event) => {
+        if (channel.cid === event.cid) setRead(channel);
+      };
 
-    const { unsubscribe } = client.on('notification.mark_read', handleEvent);
-    return unsubscribe;
-  }, [channel.cid, client, copyChannelState]);
+      const { unsubscribe } = client.on('notification.mark_read', handleEvent);
+      return unsubscribe;
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [channel.cid, client, setRead],
```

Comment:
Please fix the hook deps here

Reply:
I am not a fan of passing the entire channel object as a dep to the hook. That is why I used `channel.cid`, and hence it complains. Do you have a better way of handling it?